numpy/core/src/npymath/npy_math_internal.h.src

```diff
@@ -397,191 +292,170 @@ NPY_INPLACE double npy_log2(double x)
  * instead test for the macro, but I am lazy to do that for now.
  */
 
+
+/*
+ * Decorate all the math functions which are available on the current platform
+ */
+
 /**begin repeat
- * #type = npy_longdouble, npy_float#
- * #TYPE = NPY_LONGDOUBLE, FLOAT#
- * #c = l,f#
- * #C = L,F#
+ * #type = npy_longdouble, npy_double, npy_float#
+ * #TYPE = LONGDOUBLE, DOUBLE, FLOAT#
+ * #c = l,,f#
+ * #C = L,,F#
+ */
+#undef NPY__FP_SFX
+#if NPY_SIZEOF_@TYPE@ == NPY_SIZEOF_DOUBLE
+    #define NPY__FP_SFX(X) X
+#else
+    #define NPY__FP_SFX(X) NPY_CAT(X, @c@)
+#endif
+/*
+ * On arm64 macOS, there's a bug with sin, cos, and tan where they don't
+ * raise "invalid" when given INFINITY as input.
  */
+#if defined(__APPLE__) && defined(__arm64__)
+#define WORKAROUND_APPLE_TRIG_BUG 1
+#else
+#define WORKAROUND_APPLE_TRIG_BUG 0
+#endif
 
 /**begin repeat1
- * #kind = sin,cos,tan,sinh,cosh,tanh,fabs,floor,ceil,rint,trunc,sqrt,log10,
- *         log,exp,expm1,asin,acos,atan,asinh,acosh,atanh,log1p,exp2,log2#
- * #KIND = SIN,COS,TAN,SINH,COSH,TANH,FABS,FLOOR,CEIL,RINT,TRUNC,SQRT,LOG10,
- *         LOG,EXP,EXPM1,ASIN,ACOS,ATAN,ASINH,ACOSH,ATANH,LOG1P,EXP2,LOG2#
+ * #kind = sin,cos,tan,sinh,cosh,tanh,fabs,floor,ceil,rint,trunc,log10,
+ *         log,expm1,asin,acos,atan,asinh,acosh,atanh,log1p,exp2#
```

Comment:
@mattip you removed `exp,log2,sqrt` from this list; they are handled later but without the `WORKAROUND_APPLE_TRIG_BUG` handling. I'm not sure if that was on purpose?

Reply:
Yes it was on purpose. The comment about `WORKAROUND_APPLE_TRIG_BUG` says this, which only mentions `sin`, `cos`, `tan`. xref #19974 for more discussion.
> On arm64 macOS, there's a bug with sin, cos, and tan where they don't raise "invalid" when given INFINITY as input.

Reply:
Thanks, seems to work fine indeed.